### PR TITLE
Migrate to new framework defaults

### LINKED ISSUE #2617: Migrate to Rails 5 new framework defaults
Split out from #2271 

See `config/initializers/new_framework_defaults.rb` for details

- [ ] `Rails.application.config.action_controller.raise_on_unfiltered_parameters = true`
- [ ] `Rails.application.config.action_controller.per_form_csrf_tokens = false`
- [ ] `Rails.application.config.action_controller.forgery_protection_origin_check = false`
- [ ] `ActiveSupport.to_time_preserves_timezone = false`
- [ ] `Rails.application.config.active_record.belongs_to_required_by_default = false`
- [ ] `ActiveSupport.halt_callback_chains_on_return_false = true`

### PR DESCRIPTION
**Pull request by jeremyyap**
Close #2617 

The following new defaults affect our existing code

- `ActiveSupport.halt_callback_chains_on_return_false = false`

Now we need to use `throw(:abort)` to halt the callback chain

### CODE REVIEW

**Comment by allenwq** [comment]
there's another boolean introduced in rails5: 
```
# Configure SSL options to enable HSTS with subdomains. Previous versions had false.
Rails.application.config.ssl_options = { hsts: { subdomains: true } } 
```

Does it apply to us?

**Comment by jeremyyap** [comment]
Not sure why the generated file didn't include this option. Will look into it.

File: config/initializers/new_framework_defaults.rb [hunk 1]
```diff
@@ -6,20 +6,20 @@
 #
 # Read the Guide for Upgrading Ruby on Rails for more info on each option.
 
-Rails.application.config.action_controller.raise_on_unfiltered_parameters = true
+Rails.application.config.action_controller.raise_on_unfiltered_parameters = false
```
Comment (weiqingtoh) [inline config/initializers/new_framework_defaults.rb]: For this file, should we be deleting it after we know that flipping the switches doesn't break anything?

Comment (jeremyyap) [reply config/initializers/new_framework_defaults.rb]: We can't delete this yet. I think Rails 5.0 generates this file for new apps too.

https://github.com/rails/rails/pull/28469

We can remove this in 5.1 by calling `config.load_defaults` in `application.rb`

**Pull request merged**